### ISSUE TITLE
[Bug]: clangd ignoreRule always return "no diagnostics found in the next 10 lines."

### ISSUE THREAD

**Issue by wroyca** [chrisgrieser/nvim-rulebook#3]
### Bug Description

As per the title, clangd will always return "no diagnostics found in the next 10 lines" with ignoreRule() even though lookupRule() works as expected:

```lua
---@type LazyPluginSpec
return {
  "chrisgrieser/nvim-rulebook",
  event = "LspAttach",
  
  keys = {
    { 
      "<leader>i", 
      function() 
        require "rulebook".ignoreRule() 
      end 
    },
    { 
      "<leader>q", 
      function() 
        require "rulebook".lookupRule() 
      end 
    }
  },
	
  opts = {
    defaultConfig = {
      ignoreComments = {
        clangd = {
          comment = "//NOLINT(%s)",
          location = "sameLine",
        }
      },
      ruleDocs = {
        clangd = "https://clang.llvm.org/extra/clang-tidy/checks/%s.html"
      }
    }
  }
}
```

### Relevant Screenshot

![image](https://github.com/chrisgrieser/nvim-rulebook/assets/91024200/6340a93d-93eb-4263-a190-44ee4b79b7f6)

### neovim version

```Text
NVIM v0.9.2
Build type: RelWithDebInfo
LuaJIT 2.1.1692716794
```

### Make sure you have done the following

- [X] I have updated to the latest version of the plugin.

**Comment by chrisgrieser** [comment]
Could you move your cursor next to the diagnostic, run `:lua = vim.diagnostic.get_next()` and copypaste the output here?

**Comment by chrisgrieser** [comment]
Oh, nevermind, I see the issue. `defaultConfig` is not actually part of the config, hence making your config invalid. It should be:

```lua
opts = {
    ignoreComments = {
      clangd = {
        comment = "//NOLINT(%s)",
        location = "sameLine",
      }
    },
    ruleDocs = {
      clangd = "https://clang.llvm.org/extra/clang-tidy/checks/%s.html"
    }
}
```

**Comment by wroyca** [comment]
> Oh, nevermind, I see the issue. `defaultConfig` is not actually part of the config, hence making your config invalid. It should be:
> 
> ```lua
> opts = {
>     ignoreComments = {
>       clangd = {
>         comment = "//NOLINT(%s)",
>         location = "sameLine",
>       }
>     },
>     ruleDocs = {
>       clangd = "https://clang.llvm.org/extra/clang-tidy/checks/%s.html"
>     }
> }
> ```

I've tried that, of course \;) The issue remain the same in this case - here's the lua output:

![Screenshot from 2023-10-12 15-14-02](https://github.com/chrisgrieser/nvim-rulebook/assets/91024200/28e7ec79-602c-42cd-96a3-0b042715fdcf)

**Comment by chrisgrieser** [comment]
Ah, I think the issue is that the source is `clang-tidy`, not `clang`?

**Comment by chrisgrieser** [comment]
yes, you can do `["clang-tidy"]` in that case. That method also works in other languages, too, btw (javascript, for example).

btw, I just added support built-in support for `clang-tidy`, so it should work ootb without the need for any config on your end :)

**Comment by wroyca** [comment]
> yes, you can do `["clang-tidy"]` in that case. That method also works in other languages, too, btw (javascript, for example).
TIL

> btw, I just added support built-in support for `clang-tidy`, so it should work ootb without the need for any config on your end, once you update the plugin :)

Work great! Thank!
